services/scan/src/simple_interpreter.ts

```diff
@@ -0,0 +1,211 @@
+import { interpretTemplate } from '@votingworks/ballot-interpreter-vx';
+import * as qrcodeWorker from './workers/qrcode';
+import { Interpreter as VxInterpreter } from './interpreter';
+import { mapSheet, SheetOf } from './types';
+import { saveSheetImages } from './util/save_images';
+import { ScannerLocation, SCANNER_LOCATION } from './globals';
+import { Store } from './store';
+import { pdfToImages } from './util/pdf_to_images';
+
+export interface CreateInterpreterOptions {
+  readonly electionDefinition: ElectionDefinition;
+  readonly layouts: readonly BallotPageLayoutWithImage[];
+  readonly ballotImagesPath: string;
+  readonly testMode: boolean;
+}
+
+/**
+ * Provides a simple interface for interpreting a sheet of ballot paper.
+ */
+export interface SimpleInterpreter {
+  interpret(
+    sheetId: Id,
+    sheet: SheetOf<string>
+  ): Promise<Result<SheetOf<PageInterpretationWithFiles>, Error>>;
+}
+
+function createNhInterpreter(
+  options: CreateInterpreterOptions
+): SimpleInterpreter {
+  const { electionDefinition, ballotImagesPath } = options;
+  return {
+    interpret: async (sheetId, sheet) => {
+      const result = await interpretNh(electionDefinition, sheet);
+
+      if (result.isErr()) {
+        return result;
+      }
+
+      const [frontResult, backResult] = result.ok();
+
+      const frontImages = await saveSheetImages(
+        sheetId,
+        ballotImagesPath,
+        sheet[0],
+        frontResult.normalizedImage
+      );
+      const backImages = await saveSheetImages(
+        sheetId,
+        ballotImagesPath,
+        sheet[1],
+        backResult.normalizedImage
+      );
+
+      return ok([
+        {
+          interpretation: frontResult.interpretation,
+          originalFilename: frontImages.original,
+          normalizedFilename: frontImages.normalized,
+        },
+        {
+          interpretation: backResult.interpretation,
+          originalFilename: backImages.original,
+          normalizedFilename: backImages.normalized,
+        },
+      ]);
+    },
+  };
+}
+
+function createVxInterpreter({
+  electionDefinition,
+  ballotImagesPath,
+  layouts,
+  testMode,
+}: CreateInterpreterOptions): SimpleInterpreter {
+  const vxInterpreter = new VxInterpreter({
+    electionDefinition,
+    testMode,
+    adjudicationReasons:
+      (SCANNER_LOCATION === ScannerLocation.Central
+        ? electionDefinition.election.centralScanAdjudicationReasons
+        : electionDefinition.election.precinctScanAdjudicationReasons) ?? [],
+  });
+
+  for (const layout of layouts) {
+    vxInterpreter.addHmpbTemplate(layout);
+  }
+
+  return {
+    interpret: async (sheetId, [frontPath, backPath]) => {
+      const [frontQrcodeOutput, backQrcodeOutput] =
+        qrcodeWorker.normalizeSheetOutput(
+          electionDefinition,
+          await mapSheet(
+            [frontPath, backPath],
+            qrcodeWorker.detectQrcodeInFilePath
+          )
+        );
+      return ok(
+        await mapSheet(
```

Comment:
@eventualbuddha are we good with `mapSheet` in prod or were you thinking of moving back to a worker model? I didn't look closely at the worker pool - is it using multiple processes?

Reply:
The `mapSheet` helper itself I think is useful and could be applied in a lot of places.

As for the worker pool, here are my thoughts on that:
- maybe it's fast enough without it? @benadida to consider this
- while having our own worker pool implementation has made some things easier, like making `Result` be sendable across processes, maybe we should be using [Piscina](https://github.com/piscinajs/piscina) or similar. There are probably gremlins in here as regards our use of TypeScript in dev and compiled JS in production as that's something I've had to explicitly support.

Reply:
I like the idea of trying this first without parallelizing / workers.

Reply:
the parallelizing is likely more important for central scanner anyways.